client/src/templates/Challenges/classic/desktop-layout.tsx

```diff
@@ -52,10 +92,12 @@ const DesktopLayout = (props: DesktopLayoutProps): JSX.Element => {
       case 'showPreviewPane':
         if (!showPreviewPane && showPreviewPortal) setShowPreviewPortal(false);
         setShowPreviewPane(!showPreviewPane);
+        portalWindow?.close();
         break;
       case 'showPreviewPortal':
         if (!showPreviewPortal && showPreviewPane) setShowPreviewPane(false);
         setShowPreviewPortal(!showPreviewPortal);
+        portalWindow?.close();
```

Comment:
```suggestion
        if(showPreviewPortal) portalWindow?.close();
```
It's a bit odd to be trying to close the window during an operation that opens it.